src/renderer/browser/reducers/browserReducer.js

```diff
@@ -86,6 +86,23 @@ function open(state, action) {
   const { tabs } = state;
   const { type, target } = action;
   const internal = type === INTERNAL;
+  let tabState = {};
+
+  Object.keys(tabs).forEach((key) => {
```

Comment:
We only want this to happen for internal links (when `internal` is true), not external.  I also think this logic can be extracted into a helper function for clarity.  Perhaps something like this:

```js
import { findKey } from 'lodash';

function getInternalTab(tabs, action) {
  return findKey(tabs, (tab) => tab.type === action.type && tab.target === action.target);
}

function open(state, action) {
  // snip...
  const existingSessionId = findExistingTab(tabs, action);

  if (internal && existingSessionId) {
    return {
      ...state,
      activeSessionId: existingSessionId
    };
  }
}
```